templates/vm.xml.j2

```diff
@@ -35,7 +35,18 @@
       <driver name='qemu' type='{{ volume.format | default(libvirt_volume_default_format) }}'/>
       {% if volume.type | default(libvirt_volume_default_type) == 'file' %}
       <source file='{{ volume.file_path |default(libvirt_volume_default_images_path) }}/{{ volume.name}}'/>
-      {% else %}
+      {% elif volume.type | default(libvirt_volume_default_type) == 'network' %}
+        {% if volume.auth.name %}
+        <auth username='{{ volume.auth.name }}'>
+          <secret type='{{ volume.auth.type }}' uuid='{{ volume.auth.token }}'/>
+        </auth>
+        {% endif %} {# End volume.auth.name check #}
+        {% if volume.source.name is defined %}
```

Comment:
Is the source mandatory? We use it earlier unconditionally with qemu-img.

Reply:
If the type is network source is required so i'll adjust that check.